numpy/lib/function_base.py

```diff
@@ -1616,27 +1619,45 @@ def trim_zeros(filt, trim='fb'):
     >>> np.trim_zeros(a, 'b')
     array([0, 0, 0, ..., 0, 2, 1])
 
+    >>> b = np.array(['', '', '' ,'a', 'b', 'c', '', 'b', 'a', ''])
+    >>> np.trim_zeros(b)
+    array(['a', 'b', 'c', '', 'b', 'a'], dtype='<U1')
+
     The input data type is preserved, list/tuple in means list/tuple out.
 
     >>> np.trim_zeros([0, 1, 2, 0])
     [1, 2]
 
     """
-    first = 0
-    trim = trim.upper()
-    if 'F' in trim:
-        for i in filt:
-            if i != 0.:
-                break
-            else:
-                first = first + 1
-    last = len(filt)
-    if 'B' in trim:
-        for i in filt[::-1]:
-            if i != 0.:
-                break
-            else:
-                last = last - 1
+    try:
```

Comment:
Without such ``try`` / ``except`` approach you get weird situation were one can pass a list of strings but not an array of strings:

``` python

In [1]: import numpy as np                                                                                                                             

In [2]: a = ['a', 'b', 'c', '', 'b', 'a']                                                                                              

In [3]: np.asanyarray(a, dtype=bool)                                                                                                                   
Out[3]: 
array([True,  True,  True, False,  True,  True])

In [4]: b = np.array(a)                                                                                                                                

In [5]: np.asanyarray(b, dtype=bool)                                                                                                                   
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
    ...
ValueError: invalid literal for int() with base 10: ''
```